plugins/scaffolder/src/components/ScaffolderPage/ScaffolderPage.tsx

```diff
@@ -47,24 +62,65 @@ const getTemplateCardProps = (
   };
 };
 
-export const ScaffolderPage = () => {
-  const catalogApi = useApi(catalogApiRef);
-  const errorApi = useApi(errorApiRef);
-
-  const { data: templates, isValidating, error } = useStaleWhileRevalidate(
-    'templates/all',
-    async () => {
-      const response = await catalogApi.getEntities({
-        filter: { kind: 'Template' },
-      });
-      return response.items as TemplateEntityV1alpha1[];
-    },
+export const ScaffolderPageContents = () => {
+  const styles = useStyles();
+  const {
+    loading,
+    error,
+    filteredEntities,
+    availableCategories,
+  } = useFilteredEntities();
+  const configApi = useApi(configApiRef);
+  const orgName = configApi.getOptionalString('organization.name') ?? 'Company';
+  const { isStarredEntity } = useStarredEntities();
+  const filterGroups = useMemo<ButtonGroup[]>(
+    () => [
+      {
+        name: orgName,
+        items: [
+          {
+            id: 'all',
+            label: 'All',
+            filterFn: () => true,
+          },
+        ],
+      },
+      {
+        name: 'Personal',
+        items: [
+          {
+            id: 'starred',
+            label: 'Starred',
+            icon: StarIcon,
+            filterFn: isStarredEntity,
+          },
+        ],
+      },
+    ],
+    [isStarredEntity, orgName],
+  );
+  const [search, setSearch] = useState('');
+  const [matchingEntities, setMatchingEntities] = useState(
+    [] as TemplateEntityV1alpha1[],
   );
 
+  // Match templates by search input value
   useEffect(() => {
-    if (!error) return;
-    errorApi.post(error);
-  }, [error, errorApi]);
+    const searchUppercase = search.toUpperCase();
+    if (search.length === 0) {
+      return setMatchingEntities(filteredEntities);
+    }
+    // Match search by title|tags
+    return setMatchingEntities(
+      filteredEntities.filter(template => {
+        const { title, tags } = template.metadata;
+        return (
+          `${title}`.toUpperCase().indexOf(searchUppercase) !== -1 ||
+          tags?.join('').toUpperCase().indexOf(searchUppercase) !== -1
+        );
+      }),
+    );
```

Comment:
This is nitpicking but I think it makes it more clear if you declare a `matchSearchQuery` function that takes in as arguments the title and tags. You can then just apply it to the `filter`, e.g something along the lines of

```Javascript
const matchesQuery = (metadata, query) => 
         `${metadata.title}`.toUpperCase().indexOf(query) !== -1 ||
          metadata.tags?.join('').toUpperCase().indexOf(query) !== -1
useEffect(()...
const matches = filteredEntities.filter(template => matchesQuery(template.metadata, searchUppercase))
```

This might help remove the comments. I'd personally also use a map to first transform the metadata to uppercase but that's personal preference.